cmd/orchestrator/orchestrator.go

```diff
@@ -0,0 +1,175 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//           $$\                           $$\ $$\   $$\
+//           $$ |                          $$ |\__|  $$ |
+//  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+// $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+// $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+// $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+// \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+//  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+//
+// This file is part of Clouditor Community Edition.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"clouditor.io/clouditor/v2/internal/config"
+	"clouditor.io/clouditor/v2/logging/formatter"
+	"clouditor.io/clouditor/v2/persistence"
+	"clouditor.io/clouditor/v2/persistence/gorm"
+	"clouditor.io/clouditor/v2/persistence/inmemory"
+	"clouditor.io/clouditor/v2/server"
+	"clouditor.io/clouditor/v2/server/rest"
+	"clouditor.io/clouditor/v2/service"
+	service_orchestrator "clouditor.io/clouditor/v2/service/orchestrator"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	srv                 *server.Server
+	orchestratorService *service_orchestrator.Service
+	db                  persistence.Storage
+
+	log *logrus.Entry
+)
+
+var engineCmd = &cobra.Command{
+	Use:   "engine",
+	Short: "engine launches the Clouditor Engine",
+	Long:  "Clouditor Engine is the main component of Clouditor. It is an all-in-one solution of several microservices, which also can be started individually.",
+	RunE:  doCmd,
+}
+
+func init() {
+	log = logrus.WithField("component", "orchestrator-grpc")
+	log.Logger.Formatter = formatter.CapitalizeFormatter{Formatter: &logrus.TextFormatter{ForceColors: true}}
+	cobra.OnInitialize(config.InitConfig)
+
+	engineCmd = config.InitCobra(engineCmd)
+}
+
+func doCmd(_ *cobra.Command, _ []string) (err error) {
+	var (
+		rt, _ = service.GetRuntimeInfo()
+		level logrus.Level
+	)
+
+	fmt.Printf(`
+           $$\                           $$\ $$\   $$\
+           $$ |                          $$ |\__|  $$ |
+  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+ $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+ $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+ $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+ \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+ 
+  Clouditor Orchestrator Service Version %s
+  `, rt.VersionString())
+	fmt.Println()
+
+	level, err = logrus.ParseLevel(viper.GetString(config.LogLevelFlag))
```

Comment:
I wonder if we could move more of these functions in the internal class, e.g. everything between line 73 and 117 at least and probably the same further down as well. Otherwise its really a lot of duplicated code we have to maintain if we change things.